File: android/src/main/kotlin/com/shareinvest/flutter_kakao_maps/model/CompassOptions.kt
package com.shareinvest.flutter_kakao_maps.model

import com.kakao.vectormap.PoiScale
import org.json.JSONObject

data class CompassOptions(
    var enabled: Boolean,
    var position: KakaoMapPosition,
)

fun JSONObject.toCompassOptions(): CompassOptions {
    return CompassOptions(
        this.getBoolean("enabled"),
        this.getJSONObject("position").toKakaoMapPosition(),
    )
}

File: ios/flutter_kakao_maps/Sources/flutter_kakao_maps/model/CompassOptions.swift
import Foundation

struct CompassOptions {
    var enabled: Bool
    var position: KakaoMapPosition
}

extension NSDictionary {
    func toCompassOptions() -> CompassOptions {
        return CompassOptions(
            enabled: self["enabled"] as! Bool,
            position: (self["position"] as! NSDictionary).toKakaoMapPosition()
        )
    }
}
